grpc-interface-support/src/cpp.py

```diff
@@ -18,19 +18,323 @@
+            ),
+            templates.CopySpec(
+                "ServiceNameServiceServerFactory.cc",
+                f"{self.__get_source_dir()}/{to_camel_case(service_name)}ServiceServerFactory.cc",
+            ),
+        ]
+
+    def install_package(self) -> None:
+        conan_helper.move_generated_sources(
+            self.__package_directory_path,
+            self.__package_directory_path,
+            self.__get_include_dir(),
+            self.__get_source_dir(),
+        )
+
+        files_to_copy = [
+            templates.CopySpec(source_path="CMakeLists.txt"),
+            templates.CopySpec(source_path="conanfile.py"),
+        ]
+
+        variables = self.__get_template_variables()
+
+        cmake_headers = [
+            os.path.join(self.__get_include_dir(), file)
+            for file in os.listdir(
+                os.path.join(self.__package_directory_path, self.__get_include_dir())
+            )
+        ]
+
+        cmake_sources = [
+            os.path.join(self.__get_source_dir(), file)
+            for file in os.listdir(
+                os.path.join(self.__package_directory_path, self.__get_source_dir())
+            )
+        ]
+
+        variables["cmake_headers"] = "\n\t".join(cmake_headers)
+        variables["cmake_sources"] = "\n\t".join(cmake_sources)
+
+        templates.copy_templates(
+            get_template_dir(),
+            self.__package_directory_path,
+            files_to_copy,
+            variables,
+        )
+
+        conan_helper.export_conan_project(self.__package_directory_path)
+
+    def __transform_header_stub_code(self, lines: List[str]) -> List[str]:
+        service_name = self.__proto_file_handle.get_service_name()
+        service_class_name = to_camel_case(service_name) + "Service"
+
+        result: List[str] = []
+
+        for line in lines:
+            line = line.rstrip().replace("Service", service_class_name)
+            if line.lstrip().startswith("virtual"):
+                result.append(line.replace("virtual ", "").replace(";", " override;"))
+
+        return result
+
+    def __transform_source_stub_code(self, lines: List[str]) -> List[str]:
+        service_name = self.__proto_file_handle.get_service_name()
+        service_class_name = to_camel_case(service_name) + "Service"
+
+        result: List[str] = []
+
+        for line in lines:
+            result.append(line.replace(f"{service_name}::Service", service_class_name))
+
+        return result
+
+    def __create_or_update_service_header(self) -> None:
+        header_stub_code = GrpcCodeExtractor(
+            self.__proto_file_handle, self.__package_directory_path
+        ).get_header_stub_code(self.__get_include_dir())
+
+        header_stub_code = self.__transform_header_stub_code(header_stub_code)
+
+        app_source_dir = os.path.join(get_workspace_dir(), "app", "src")
+        service_header_file_name = (
+            f"{self.__proto_file_handle.get_service_name()}ServiceImpl.h"
```

Comment:
since you already defined the variables please use it

Reply:
Done